hpOneView/resources/security/login_details.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+###
+# (C) Copyright (2012-2017) Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+###
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from future import standard_library
+
+standard_library.install_aliases()
+
+
+from hpOneView.resources.resource import ResourceClient
+
+
+class LoginDetails(object):
+    """
+    Login API client.
+
+    """
+    URI = '/rest/logindetails'
+    DEFAULT_VALUES = {
+        '500': {'type': 'LoginDetails'}
+    }
+
+    def __init__(self, con):
+        self._connection = con
+        self._client = ResourceClient(con, self.URI)
+
+    def get(self, id_or_uri):
+        """
+        Retrieve login details by its uri.
+
+        Args:
+            id_or_uri: session id (or uri)
+
+        Returns:
+            dict: The login details.
+
+        """
+
+        login = self._client.get(id_or_uri)
```

Comment:
You could swap lines 63-64 for:
```python
return self._client.get(self.URI)
```

That'd mean that you no longer need to receive an `id_or_uri` from the user, and you could change the method to:

```python
def get(self):
```

Perhaps modifying the name would be better to avoid confusion with other `get` methods, since this one would not accept the id or uri as it is default. Perhaps calling it `get_login_details` or something similar.